src/lib.rs

```diff
@@ -472,25 +511,38 @@ impl Transitions for Dense {
 ///
 /// This can use enormous amounts of memory when there are many patterns,
 /// but matching is very fast.
-#[derive(Clone, Debug)]
-pub struct Sparse(Vec<StateIdx>);
+pub struct Sparse([StateIdx; 256]);
+
+impl Clone for Sparse {
+    fn clone(&self) -> Sparse {
+        Sparse(self.0)
+    }
+}
+
+impl fmt::Debug for Sparse {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_tuple("Sparse").field(&&self.0[..]).finish()
+    }
+}
 
 impl Transitions for Sparse {
     fn new(_: u32) -> Sparse {
-        Sparse(vec![0; 256])
+        Sparse([0; 256])
     }
 
     #[inline]
     fn goto(&self, b: u8) -> StateIdx {
-        self.0[b as usize]
+        unsafe { *self.0.get_unchecked(b as usize) }
```

Comment:
Again. Please justify `unsafe`. I am super super skeptical of just going around removing bounds checks. They so rarely have an observable impact.

Reply:
It helped on debug builds when `Sparse` was a `Vec` and since taking a byte as index guaranteed that it would be in range I figured it were a small enough use to be ok.

Since it is a fixed size array now it does not matter either way so I removed it (it gets eliminated either way it seems).